### DISCUSSION TITLE
How to display "user already exists" is user tries to sign up again?

### DISCUSSION THREAD

**Discussion by shawnesquivel** [clerk/javascript#2527] [Support request - DO NOT OPEN HERE]
My SignIn and SignUp features work as expected.

Let's say I have Signed Up with Google Auth, so my user is now in the database.

![image](https://github.com/clerk/javascript/assets/94336773/f836bfd7-46a0-462b-b6c0-938fde782aa2)

Now, when I try to Sign Up again using the `<SignUp/>` component (https://clerk.com/docs/components/authentication/sign-up), we would expect an error because I already exist in the User database. Most of the methods from the docs assume that SignIn or SignUp was successful.

The application seems to understand this, and SignUp redirects me to this URL: `http://localhost:3000/sign-up/sso-callback?redirect_url=http%3A%2F%2Flocalhost%3A3000%2F`

My question is, is there some way to display this information to my user? E.g. "you already have an account, please try Signing In"? My console seems to give one hint, but because Sign Up is abstracted, I'm not sure where this SSO callback is being performed.

```
GET http://localhost:3000/sign-up/sso-callback?redirect_url=http%3A%2F%2Flocalhost%3A3000%2F 404 (Not Found)
```

**Recording**
https://github.com/clerk/javascript/assets/94336773/5a319b48-c7a1-40dd-8ae5-25ce8cac90d6

**Comment by imran-mirza79** [comment]
> My SignIn and SignUp features work as expected.
> 
> Let's say I have Signed Up with Google Auth, so my user is now in the database.
> 
> ![image](https://github.com/clerk/javascript/assets/94336773/f836bfd7-46a0-462b-b6c0-938fde782aa2)
> 
> Now, when I try to Sign Up again using the `<SignUp/>` component (https://clerk.com/docs/components/authentication/sign-up), we would expect an error because I already exist in the User database. Most of the methods from the docs assume that SignIn or SignUp was successful.
> 
> The application seems to understand this, and SignUp redirects me to this URL: `http://localhost:3000/sign-up/sso-callback?redirect_url=http%3A%2F%2Flocalhost%3A3000%2F`
> 
> My question is, is there some way to display this information to my user? E.g. "you already have an account, please try Signing In"? My console seems to give one hint, but because Sign Up is abstracted, I'm not sure where this SSO callback is being performed.
> 
> ```
> GET http://localhost:3000/sign-up/sso-callback?redirect_url=http%3A%2F%2Flocalhost%3A3000%2F 404 (Not Found)
> ```
> 
> **Recording**
> https://github.com/clerk/javascript/assets/94336773/5a319b48-c7a1-40dd-8ae5-25ce8cac90d6
> 

Hey, I've a similar requirement. Did you get any workaround for this?

**Comment by dima-jabali** [comment]
Any news on this??

**Comment by jescalan** [comment]
https://github.com/orgs/clerk/discussions/3829
